src/api/java/baritone/api/utils/Helper.java

```diff
@@ -74,20 +104,40 @@ default void logDebug(String message) {
             //System.out.println(message);
             return;
         }
-        logDirect(message);
+        // We won't log debug chat into toasts
+        // Because only a madman would want that extreme spam -_-
+        logDirect(message, false);
     }
 
     /**
      * Send components to chat with the [Baritone] prefix
      *
+     * @param logAsToast Whether to log as a toast notification
      * @param components The components to send
      */
-    default void logDirect(ITextComponent... components) {
+    default void logDirect(boolean logAsToast, ITextComponent... components) {
         ITextComponent component = new TextComponentString("");
-        component.appendSibling(getPrefix());
-        component.appendSibling(new TextComponentString(" "));
+        if (!logAsToast) {
+            // If we are not logging as a Toast
+            // Append the prefix to the base component line
+            component.appendSibling(getPrefix());
+            component.appendSibling(new TextComponentString(" "));
+        }
         Arrays.asList(components).forEach(component::appendSibling);
-        Minecraft.getMinecraft().addScheduledTask(() -> BaritoneAPI.getSettings().logger.value.accept(component));
+        if (logAsToast) {
+            logToast(getPrefix(), component);
```

Comment:
Not thread safe - pathing thread calls this sometimes. There was an `addScheduledTask` for a reason lol